pkg/controller/rotation/handle_secrets_rotate.go

```diff
@@ -189,9 +189,19 @@ func (c *Controller) rotateSecret(ctx context.Context, typ database.SecretType,
 		existing = append(existing, created...)
 	}
 
+	// Find the "newest" secret that has been created. Since ListSecretsForType
+	// sorts by usable, it's possible that a new secret exists but is not yet
+	// active. That would be at the end of the list.
+	latestSecretCreatedAt := time.Time{}
+	for _, secret := range existing {
+		if secret.CreatedAt.UTC().After(latestSecretCreatedAt) {
+			latestSecretCreatedAt = secret.CreatedAt.UTC()
+		}
+	}
+
 	// If there is no latest secret, or if the latest secret has existed for
 	// longer than the minimum age, create a new one.
-	if minTTL > 0 && (len(existing) == 0 || now.Sub(existing[0].CreatedAt) > minTTL) {
+	if minTTL > 0 && now.Sub(latestSecretCreatedAt) > minTTL {
```

Comment:
I think this is easier to read if it was

`now.Add(minTTL).After(lastScretCreatedAt)`

but those are the same thing.

Reply:
or Before I guess in this case...

Reply:
time is reletive